src/modules/Categories.js

```diff
@@ -0,0 +1,227 @@
+import Category from './Category.js';
+import Meal from './Meal.js';
+
+export default class Categories {
+  myId;
+
+  constructor(container, buttons, popupComments) {
+    this.container = container;
+    this.buttons = buttons;
+    this.popupComments = popupComments;
+    this.categories = [];
+    this.currentCategory = '';
+    this.myId = 'NJoITshHrkNNBBBobyAA';
+  }
+
+  load(name) {
+    return new Promise((resolve, reject) => {
+      const added = this.categories.find((cat) => cat.name === name);
+      const isAdded = added !== undefined;
+      if (isAdded) {
+        this.displayMeals(added.meals, name);
+        this.changeCurrentButton(name);
+        this.currentCategory = name;
+        resolve();
+        return;
+      }
+
+      const category = new Category(name);
+      category
+        .getMeals()
+        .then((meals) => {
+          this.categories.push(category);
+          this.displayMeals(meals, name);
+          this.changeCurrentButton(name);
+          this.currentCategory = name;
+          resolve();
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+  }
+
+  getComments = async (id) => {
+    const url = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${this.myId}/comments?item_id=${id}`;
+    const result = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    return result;
+  };
+
+  initApp = async () => {
+    const url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/';
+    const result = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    result.then((res) => { res.text().then((a) => console.log(a)); });
+  };
+
+  addComment = async (itemId, name, comment) => {
+    const url = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${this.myId}/comments`;
+    console.log(url);
+    const result = fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(
+        {
+          item_id: itemId,
+          username: name,
+          comment,
+        },
+      ),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+
+    const promise1 = result.then((res) => { res.text().then((a) => console.log(a)); });
+    await promise1;
```

Comment:
You're storing the resulting promise from the fetch into a variable that later on is just used to console.log the data. If you're not going to do anything else with the data than just than just printing it you shouldn't wait for the promise to be fulfilled. You can remove the await from the function and the promise1 variable, also the result variable won't be used anymore. And if you're not using await you can also make this a normal function instead of using async.